### PR TITLE
VACMS 2452 Step By Step Component Alert Selection

### PR DESCRIPTION
**Pull request by kaise-lafrai**
## Description

See #2452 

## QA steps

As a content editor
1. Login
2. Navigate here: `admin/structure/paragraphs_type`
 - [x] Confirm Alert Single, Non-reusable Alert paragraph types exist with structure outline below
        -New paragraph type "Non-reusable Alert" | machine name non_reusable_alert
            -fields (same fields that are presented for a non-reusable Alert in "Alert" paragraph type):
              -Alert Heading | field_alert_heading
              -Alert Content | field_va_paragraphs
               -Alert Type | field_alert_type
        -New paragraph type "Alert (single)" | machine name alert_single
          -fields
            - Alert Selection | field_alert_selection | three options "Reusable, "Non-reusable","None"
            - Reusable alert | field_alert_block_reference
            - Non-reusable Alert | field_alert_non_reusable_reference | reference to "Non-reusable Alert" paragraph
 - [x] Confirm updates to Step paragraph type:
      -"Alert" field_alert to "Step" paragraph, below WYSIWYG. Field can reference only paragraphs of type "Alert (single)"

3. Navigate to Alert section on form to confirm field behaviors `/node/add/step_by_step`
- [x] None is selected when creating new content by default
- [x] Reusable and Non-reusable fields are not visible until either Reusable or Non-reusable Alert Selections are selected.
- [x] If Reusable is selected and an entity reference is added, all alert selection radio buttons are disabled.
- [x] If Non-reusable is selected and a Non-reusable alert is added, all alert selection radio buttons are disabled.
- [x] When removing either the Reusable or Non-reusable alert, all the fields are then re-enabled.
- [x] Confirm same behaviors exist when adding new steps and alerts.
- [x] Confirm same behaviors exist when removing all steps or behaviors and re-adding a step and alert.
- [x] Save the node and confirm that the behaviors apply when returning to edit the node.

## Definition of Done
- [ ] Product release are written (or in progress), if required.
- [ ] Documentation has been updated, if applicable.

### CODE REVIEW

File: config/sync/menu_breadcrumb.settings.yml [hunk 1]
```diff
@@ -59,62 +59,62 @@ menu_breadcrumb_menus:
     weight: -1
     taxattach: 0
     langhandle: 0
-  va-jackson-health-care:
```
Comment (swirtSJW) [inline config/sync/menu_breadcrumb.settings.yml:62]: I'm concerned that this is being removed and not added.   This is not just a simple re-ordering, this is a loss.

File: config/sync/menu_breadcrumb.settings.yml [hunk 2]
```diff
@@ -59,62 +59,62 @@ menu_breadcrumb_menus:
...
     weight: 0
     taxattach: 0
     langhandle: 0
-  va-minneapolis-health-care:
+  va-martinsburg-health-care:
     enabled: 1
     weight: 0
     taxattach: 0
     langhandle: 0
-  va-houston-health-care:
```
Comment (swirtSJW) [inline config/sync/menu_breadcrumb.settings.yml:102]: Also a loss with no addition.

File: config/sync/menu_breadcrumb.settings.yml [hunk 3]
```diff
@@ -59,62 +59,62 @@ menu_breadcrumb_menus:
...
     weight: 0
     taxattach: 0
     langhandle: 0
-  va-hampton-health-care:
+  va-maine-health-care:
     enabled: 1
     weight: 0
     taxattach: 0
     langhandle: 0
-  va-gulf-coast-health-care:
```
Comment (swirtSJW) [inline config/sync/menu_breadcrumb.settings.yml:112]: Another loss.

File: config/sync/menu_breadcrumb.settings.yml [hunk 4]
```diff
@@ -124,92 +124,37 @@ menu_breadcrumb_menus:
...
-    enabled: 1
-    weight: 0
-    taxattach: 0
-    langhandle: 0
-  va-salt-lake-city-health-care:
-    enabled: 1
-    weight: 0
-    taxattach: 0
-    langhandle: 0
-  va-san-francisco-health-care:
```
Comment (swirtSJW) [inline config/sync/menu_breadcrumb.settings.yml:187]: loss

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 5]
```diff
@@ -396,6 +396,50 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
...
+ * @param array $element
+ *   The widget.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ * @param array $context
+ *   The context.
+ */
+function _va_gov_backend_modify_alert_single_states(array &$element, FormStateInterface &$form_state, array &$context) {
+  // Toggle Reusable & Non-reusable Alert_Single subform field on alert select.
+  if ($element['#paragraph_type'] == 'alert_single') {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:420]: minor: strict comparison preferred unless there is reason not to use strict.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 6]
```diff
@@ -396,6 +396,50 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
...
+    $element_path = $element['subform']['#parents'];
+    $parent_field = array_shift($element_path);
+    $element_path_string = implode('][', $element_path);
+
+    if (!empty($parent_field) && !empty($element_path_string)) {
+      $selector = ':input[name="' . $parent_field . '[' . $element_path_string . '][field_alert_selection]"]';
+
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:427]: minor: comments describing the rationale are always helpful deciphering this 6 months from now.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 5 above)
Comment (kaise-lafrai) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:420]: I will change to ===

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 6 above)
Comment (kaise-lafrai) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:427]: I can add a comment.

**Comment by kaise-lafrai** [comment]
@swirtSJW @oksana-c Ready for review.

**Pull request merged**